src/main/java/com/labsynch/labseer/domain/DDictValue.java

```diff
@@ -568,6 +569,45 @@ public static TypedQuery<DDictValue> findDDictValuesByLsTypeEqualsAndLsKindEqual
         return q;
     }
 
+    public static List<DDictValue> findDDictValuesByLsTypeEqualsAndLsKindEqualsAndLabelTextSearch(String lsType, String lsKind, String labelText, Integer maxHits) {
+        if (lsType == null || lsType.length() == 0)
+            throw new IllegalArgumentException("The lsType argument is required");
+        if (lsKind == null || lsKind.length() == 0)
+            throw new IllegalArgumentException("The lsKind argument is required");
+        EntityManager em = DDictValue.entityManager();
+
+        // Format the search text to match any word in the label text
```

Comment:
Below is all the magic.  Split and join all search terms and use to_tsvector serching to match.  I found this to work very well in practice.  The one gotcha is that the ts_rank order by was not perfect.  I wanted to search e.g. "ec50" but it was ranking things like "BLAH-phosphorylated EC50" higher.  Adding a preferrential order by case insenstive exact match made the search function how I wanted.